grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcHealthCheckService.java

```diff
@@ -0,0 +1,314 @@
+ * the status of the gRPC service from the response by registering it to grpcServiceHealthCheckers.
+ * Note: The suggested format of service name is package_names.ServiceName
+ * If a server is healthy, returns SERVING.
+ * For more details, please refer to the following URL.
+ *
+ * @see <a href="https://github.com/grpc/grpc/blob/master/doc/health-checking.md">GRPC Health Checking Protocol</a>
+ * @see GrpcHealthCheckServiceBuilder
+ */
+@UnstableApi
+public final class GrpcHealthCheckService extends HealthImplBase {
+
+    private static final Logger logger = LoggerFactory.getLogger(GrpcHealthCheckService.class);
+
+    private static final HealthCheckResponse SERVING_RESPONSE =
+            HealthCheckResponse.newBuilder()
+                               .setStatus(ServingStatus.SERVING)
+                               .build();
+
+    private static final HealthCheckResponse NOT_SERVING_RESPONSE =
+            HealthCheckResponse.newBuilder()
+                               .setStatus(ServingStatus.NOT_SERVING)
+                               .build();
+
+    private static final HealthCheckResponse SERVICE_UNKNOWN_RESPONSE =
+            HealthCheckResponse.newBuilder()
+                               .setStatus(ServingStatus.SERVICE_UNKNOWN)
+                               .build();
+
+    /**
+     * Returns a newly created {@link GrpcHealthCheckService}
+     * with the specified {@link ListenableHealthChecker}s.
+     */
+    public static GrpcHealthCheckService of(ListenableHealthChecker... healthCheckers) {
+        return builder().checkers(healthCheckers).build();
+    }
+
+    /**
+     * Returns a newly created {@link GrpcHealthCheckService}
+     * with the specified {@link ListenableHealthChecker}s.
+     */
+    public static GrpcHealthCheckService of(Iterable<? extends ListenableHealthChecker> healthCheckers) {
+        return builder().checkers(healthCheckers).build();
+    }
+
+    /**
+     * Returns a new builder which builds a new {@link GrpcHealthCheckService}.
+     */
+    public static GrpcHealthCheckServiceBuilder builder() {
+        return new GrpcHealthCheckServiceBuilder();
+    }
+
+    private static HealthCheckResponse getHealthCheckResponse(ServingStatus status) {
+        if (status == ServingStatus.SERVING) {
+            return SERVING_RESPONSE;
+        } else if (status == ServingStatus.NOT_SERVING) {
+            return NOT_SERVING_RESPONSE;
+        } else if (status == ServingStatus.SERVICE_UNKNOWN) {
+            return SERVICE_UNKNOWN_RESPONSE;
+        } else {
+            throw new IllegalArgumentException("Invalid status:" + status);
+        }
+    }
+
+    private static void setHealthUpdateListenerToServerHealthCheckers(
+            Consumer<String> healthCheckUpdateListener,
+            Collection<ListenableHealthChecker> listenableHealthCheckers) {
+        listenableHealthCheckers.forEach(
+                lhc -> lhc.addListener(healthChecker -> healthCheckUpdateListener.accept("")));
```

Comment:
nit; what do you think of defining a separate variable for ""?

i.e.
```suggestion
                lhc -> lhc.addListener(healthChecker -> healthCheckUpdateListener.accept(EMPTY_SERVICE)));
```

Reply:
It is easy to read.